### PR TITLE
Managing articles by user

### PR DESCRIPTION
**Pull request by Kriz1618**
Listing articles by user
Deleting articles by user
Updating articles by user

### CODE REVIEW

File: articles/models.py [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
 from django.db import models
+from django.utils.timezone import now
 
 
 class Article(models.Model):
     title = models.CharField(max_length=200)
     body = models.TextField()
     image = models.URLField(max_length=200)
-    created_at = models.DateTimeField(auto_now_add=True)
+    created_at = models.DateTimeField(default=now)
```
Comment (dcousin24) [inline articles/models.py]: Check the `auto_add` and the `auto_add_now` for this two fields

File: articles/views.py [hunk 2]
```diff
@@ -19,4 +22,24 @@ class ArticleViewSet(viewsets.ModelViewSet):
...
+    ordering_fields = ['title', 'created_at']
+
+    def get_queryset(self):
+        if self.action == 'list':
+            return self.queryset.filter(author=self.request.user)
+        return self.queryset
+
+    def perform_create(self, serializer):
+        serializer.save(author=self.request.user)
+
```
Comment (dcousin24) [inline articles/views.py:31]: These methods shouldn't be handled like this. If you directly return the only the articles that are related to the request.user, the update and delete won't find it and the endpoint will return 404

File: users/views.py [hunk 3]
```diff
@@ -45,13 +54,14 @@ def get_serializer(self, *args, **kwargs):
         return self.serializer_class(*args, **kwargs)
 
 
-class Logout(GenericAPIView):
+class LogoutView(GenericAPIView):
     permission_classes = [IsAuthenticated]
+    authentication_classes = [JWTAuthentication]
```
Comment (dcousin24) [inline users/views.py]: Wasn't this set in the DRF default authentication classes?

File: articles/views.py (hunk 2 above)
Comment (Kriz1618) [reply articles/views.py:31]: the perform methods were removed

File: users/views.py (hunk 3 above)
Comment (Kriz1618) [reply users/views.py]: It wasn't necessary

**Review by dcousin24 — APPROVED** [review]

**Pull request merged**